### PR TITLE
New SLE 15 rule set_nftables_table

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _New SLE 15 rule which covers CIS requirement 3.5.2.4 Ensure a table exists (Automated)_

#### Rationale:

- At the moment CIS profile for SLE 15 does not include this rule.

### CODE REVIEW

File: linux_os/guide/system/network/network-nftables/set_nftables_table/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+{{{ ansible_instantiate_variables("var_nftables_family") }}}
+{{{ ansible_instantiate_variables("var_nftables_table") }}}
+
+- name: Collect existing nftables
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/set_nftables_table/ansible/shared.yml:10]:
```suggestion
- name: Collect Existing Nftables 
```

File: linux_os/guide/system/network/network-nftables/set_nftables_table/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# disruption = low
+
+{{{ ansible_instantiate_variables("var_nftables_family") }}}
+{{{ ansible_instantiate_variables("var_nftables_table") }}}
+
+- name: Collect existing nftables 
+  ansible.builtin.shell: nft list tables
+  register: existing_nftables
+
+- name: Set nftable table
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/set_nftables_table/ansible/shared.yml:14]:
```suggestion
- name: Set Nftable Table
```

File: linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+description: |-
+   Tables in nftables hold chains. Each table only has one address family and only applies 
+   to packets of this family. Tables can have one of six families.
+
+rationale: |-
+    Nftables doesn't have any default tables. Without a table being build, nftables will not filter
+    network traffic.
+    <b>Impact</b>Adding rules to a running nftables can cause loss of connectivity to the system
+
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml:15]: Please add warning like [this](https://github.com/ComplianceAsCode/content/blob/1b538dfc48ab5bd43f8d9d9a1b6823707dcdbf71/linux_os/guide/services/ssh/ssh_server/sshd_use_strong_rng/rule.yml#L55) about risk of losing connectivity to the system.

File: linux_os/guide/system/network/network-nftables/var_nftables_family.var [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    is a separate tool for each level: iptables, ip6tables, arptables, ebtables. 
+    With nftables the multiple networking levels are abstracted into families, 
+    all of which are served  by the single tool nft. 
+    <tt>ip</tt>Tables of this family see IPv4 traffic/packets. 
+    <tt>ip6</tt>Tables of this family see IPv6 traffic/packets.
+    <tt>inet</tt>Tables of this family see both IPv4 and IPv6 traffic/packets, 
+    simplifying dual stack support. 
+    <tt>arp</tt>Tables of this family see ARP-level (i.e, L2) traffic, before 
+    any L3 handling is done by the kernel. 
+    <tt>bdidge</tt>Tables of this family see traffic/packets traversing bridges
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/var_nftables_family.var:17]:
```suggestion
    <tt>bridge</tt>Tables of this family see traffic/packets traversing bridges 
```

File: linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+prodtype: sle15
+
+title: 'Ensure a table exists for nftables'
+
+description: |-
+   Tables in nftables hold chains. Each table only has one address family and only applies 
+   to packets of this family. Tables can have one of six families.
+
+rationale: |-
+    Nftables doesn't have any default tables. Without a table being build, nftables will not filter
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml:12]:
```suggestion
    Nftables doesn't have any default tables. Without a table being built, nftables will not filter
```

File: linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+severity: medium
+
+identifiers:
+    cce@sle15: CCE-92569-3
+
+references:
+    cis@sle15: 3.5.2.4
+
+ocil_clause: 'nftables table does not exist'
```
Comment (Mab879) [inline linux_os/guide/system/network/network-nftables/set_nftables_table/rule.yml:24]:
```suggestion
ocil_clause: 'a nftables table does not exist'
```

File: controls/cis_sle15.yml [hunk 7]
```diff
@@ -1051,7 +1051,14 @@ controls:
     levels:
       - l1_server
       - l1_workstation
-    status: manual # rule is missing
+    status: automated
+    notes: 
+       The audit (OVAL check) cannot be automated,
```
Comment (Mab879) [inline controls/cis_sle15.yml:1056]: Just a side note, if you are willing to not be SCAP compliant SCE (script check engine) content could be used on this rule to automate the check.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR. I have a few suggestions. Please also add Automatus tests for this rule as well.

**Comment by rumch-se** [comment]
Hello @Mab879 
Thank you for your feedback.
I have done the corrections proposed.
Have a nice day
Rumen

File: linux_os/guide/system/network/network-nftables/var_nftables_table.var [hunk 8]
```diff
@@ -0,0 +1,18 @@
+documentation_complete: true
+
+title: 'Nftables Tables'
+
```
Comment (marcusburghardt) [inline linux_os/guide/system/network/network-nftables/var_nftables_table.var:4]:
```suggestion
```

**Review by Mab879 — APPROVED** [review]
Thanks for the PR!

**Comment by Mab879** [comment]
@teacup-on-rockingchair can you please take a look at this?

**Comment by teacup-on-rockingchair** [comment]
> @teacup-on-rockingchair can you please take a look at this?

LGTM, not sure what I can add, besides maybe a clause on nftables package in the remediation for bash:
```
{{{ bash_package_install("nftables") }}}

``` 
and ansible:

```
- name: "Ensure nftables is installed"
  package:
    name: "{{ item }}"
    state: present
  with_items:
    - nftables
```

otherwise on  my broken SLE15 setup the remediation passes, assuming nftables commands work, when there is no nft command at all.

**Comment by Mab879** [comment]
Overriding CODEOWNERS since an SUSE approver is not available currently.

**Pull request merged**